src/Features/Core/Portable/CodeFixes/Configuration/ConfigureSeverityLevelCodeFixProvider.ConfigurationUpdater.cs

```diff
@@ -0,0 +1,357 @@
+                            }
+                        }
+
+                        if (validRule)
+                        {
+                            // We found the rule in the file -- replace it with updated severity.
+                            if (isOptionBasedMatch && key.Equals(nameOpt))
+                            {
+                                // We found a rule configuration entry of option based form:
+                                //      "%option_name% = %option_value%:%severity%
+                                var textChange = new TextChange(curLine.Span, $"{key} = {optionValue}:{_severity}{commentValue}");
+                                return result.WithChanges(textChange);
+                            }
+                            else if (isSeverityBasedMatch)
+                            {
+                                // We found a rule configuration entry of severity based form:
+                                //      "dotnet_diagnostic.<%DiagnosticId%>.severity = %severity%
+                                int diagIdLength = -1;
+                                if (key.StartsWith(DiagnosticOptionPrefix, StringComparison.Ordinal) &&
+                                    key.EndsWith(DiagnosticOptionSuffix, StringComparison.Ordinal))
+                                {
+                                    diagIdLength = key.Length - (DiagnosticOptionPrefix.Length + DiagnosticOptionSuffix.Length);
+                                }
+
+                                if (diagIdLength >= 0)
+                                {
+                                    var diagId = key.Substring(
+                                        DiagnosticOptionPrefix.Length,
+                                        diagIdLength);
+                                    if (string.Equals(diagId, _diagnostic.Id, StringComparison.OrdinalIgnoreCase))
+                                    {
+                                        var textChange = new TextChange(curLine.Span, $"{key} = {_severity}{commentValue}");
+                                        return result.WithChanges(textChange);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    else if (s_headerPattern.IsMatch(curLineText.Trim()))
+                    {
+                        // We found a header entry such as '[*.cs]', '[*.vb]', etc.
+                        // Update the most recent header.
+                        var groups = s_headerPattern.Match(curLineText.Trim()).Groups;
+                        mostRecentHeader = groups[1].Value.ToString().ToLowerInvariant();
+                    }
+
+                    if (mostRecentHeader != null)
+                    {
+                        headers[mostRecentHeader] = curLine;
+                    }
+                }
+
+                return null;
+            }
```

Comment:
skipped this method.

Reply:
Note that you already reviewed majority of this code last year in the original PR https://github.com/dotnet/roslyn/pull/29332. I have tried to preserve most of the logic from that PR in the logic to add/update editorconfig entry.